drizzle/meta/0050_snapshot.json

```diff
@@ -0,0 +1,2197 @@
+          "tableTo": "posts",
+          "columnsFrom": [
+            "post_id"
+          ],
+          "columnsTo": [
+            "id"
+          ],
+          "onDelete": "cascade",
+          "onUpdate": "no action"
+        },
+        "reactions_account_id_accounts_id_fk": {
+          "name": "reactions_account_id_accounts_id_fk",
+          "tableFrom": "reactions",
+          "tableTo": "accounts",
+          "columnsFrom": [
+            "account_id"
+          ],
+          "columnsTo": [
+            "id"
+          ],
+          "onDelete": "cascade",
+          "onUpdate": "no action"
+        }
+      },
+      "compositePrimaryKeys": {
+        "reactions_post_id_account_id_emoji_pk": {
+          "name": "reactions_post_id_account_id_emoji_pk",
+          "columns": [
+            "post_id",
+            "account_id",
+            "emoji"
+          ]
+        }
+      },
+      "uniqueConstraints": {}
+    },
+    "public.reports": {
+      "name": "reports",
+      "schema": "",
+      "columns": {
+        "id": {
+          "name": "id",
+          "type": "uuid",
+          "primaryKey": true,
+          "notNull": true
+        },
+        "account_id": {
+          "name": "account_id",
+          "type": "uuid",
+          "primaryKey": false,
+          "notNull": true
+        },
+        "target_account_id": {
+          "name": "target_account_id",
+          "type": "uuid",
+          "primaryKey": false,
+          "notNull": true
+        },
+        "created": {
+          "name": "created",
+          "type": "timestamp with time zone",
+          "primaryKey": false,
+          "notNull": true,
+          "default": "CURRENT_TIMESTAMP"
+        },
+        "comment": {
+          "name": "comment",
+          "type": "text",
+          "primaryKey": false,
+          "notNull": false
+        },
+        "posts": {
+          "name": "posts",
+          "type": "uuid[]",
+          "primaryKey": false,
+          "notNull": true,
+          "default": "'{}'::uuid[]"
+        }
+      },
+      "indexes": {},
+      "foreignKeys": {
+        "reports_account_id_accounts_id_fk": {
+          "name": "reports_account_id_accounts_id_fk",
+          "tableFrom": "reports",
+          "tableTo": "accounts",
+          "columnsFrom": [
+            "account_id"
+          ],
+          "columnsTo": [
+            "id"
+          ],
+          "onDelete": "cascade",
+          "onUpdate": "no action"
+        },
+        "reports_target_account_id_accounts_id_fk": {
+          "name": "reports_target_account_id_accounts_id_fk",
+          "tableFrom": "reports",
+          "tableTo": "accounts",
+          "columnsFrom": [
+            "target_account_id"
+          ],
+          "columnsTo": [
+            "id"
+          ],
+          "onDelete": "cascade",
+          "onUpdate": "no action"
+        }
+      },
+      "compositePrimaryKeys": {},
+      "uniqueConstraints": {}
+    },
```

Comment:
This is a reasonable suggestion, but would likely require updating the report once the report has been federated outwards. No hard opinions on it.

Reply:
I think we'd better to have at least `reports.iri` for the later implementation of `reports.federated_to`.  Or it would be much better if we have both.

Reply:
@dahlia yeah, we just need to decide what this should be. Is there a good scheme for dereferencable IRIs for Activities in Hollo?

Reply:
One thing we have in Mastodon though is one report can generate many Flag activities to different servers, potentially with varying contents. See: https://github.com/mastodon/mastodon/pull/31706 where I'm currently building this.

I think what I'll end up with is having `Report` model be the local copy, and then `ReportForward` being the federated ones, with the target domains associated. So creating a report but not forwarding it, would create a Report but no ReportForward records, and creating a report and forwarding it to two servers would create one Report and one ReportForward with two domains in that forward. (ReportForward can have a comment that overrides the Report comment)

Reply:
> @dahlia yeah, we just need to decide what this should be. Is there a good scheme for dereferencable IRIs for Activities in Hollo?

Though it's not for activities, here's an instance:

<https://github.com/dahlia/hollo/blob/90f265437452b3153036e5ef67139fb38a651d0a/src/federation/index.ts#L889-L905>

Since [`Federation.setObjectDispatcher()`](https://fedify.dev/manual/object) also works with `Activity` and its subtypes, you could utilize this.  FYI you could generate a URI for this by calling [`Context.getObjectUri()`](https://fedify.dev/manual/object#constructing-object-uris).

Reply:
Cool, that's what I've done.